packages/braid-design-system/lib/hooks/useIcon/index.ts

```diff
@@ -6,20 +6,22 @@ import { OptionalTitle } from '../../components/icons/SVGTypes';
 import { PublicBoxProps } from '../../components/Box/Box';
 import { TextContext } from '../../components/Text/TextContext';
 import HeadingContext from '../../components/Heading/HeadingContext';
-import { textSize, useTextTone, UseTextProps } from '../typography';
 import { lineHeightContainer } from '../../css/lineHeightContainer.css';
 import buildDataAttributes, {
   DataAttributeMap,
 } from '../../components/private/buildDataAttributes';
+import * as typographyStyles from '../../css/typography.css';
 import * as styles from './icon.css';
 
-type IconSize = NonNullable<UseTextProps['size']> | 'fill';
+type IconSize =
+  | NonNullable<keyof typeof typographyStyles.textSizeUntrimmed>
+  | 'fill';
```

Comment:
I believe this still works as expected:

```suggestion
type IconSize = keyof typeof typographyStyles.textSizeUntrimmed | 'fill';
```